include/seqan3/alphabet/aminoacid/concept.hpp

```diff
@@ -6,47 +6,124 @@
+    // ADL
+    SEQAN3_CPO_IMPL(1, std::bool_constant<enable_aminoacid(t{})>::value)
+    // default: derived from base class or not (valid for any type)
+    SEQAN3_CPO_IMPL(0, (std::is_base_of_v<seqan3::aminoacid_empty_base, strip_type_identity_t<t>>))
+
+    //!\brief Main dispatching function.
+    template <typename alph_t>
+    static constexpr bool impl_main() noexcept
+    {
+        if constexpr (std::is_nothrow_default_constructible_v<alph_t> &&
+                      seqan3::is_constexpr_default_constructible_v<alph_t>)
+        {
+            return impl(priority_tag<2>{}, alph_t{});
+        }
+        else
+        {
+            return impl(priority_tag<2>{}, std::type_identity<alph_t>{});
+        }
+    }
+};
+
+} // namespace seqan3::detail::adl_only
+
+namespace seqan3
+{
+
+/*!\brief A trait that indicates whether a type shall model seqan3::aminoacid_alphabet.
+ * \tparam your_type Type of the argument.
  * \ingroup aminoacid
- *
  * \details
  *
- * Since an amino acid alphabet has no specific characteristics (like the complement
- * function for nucleotide alphabets), we distinguish an amino acid alphabet by
- * the seqan3::is_aminoacid unary type trait.
+ * This is an auxiliary trait that is checked by seqan3::aminoacid_alphabet to verify that a type is an amino acid.
+ * This trait should never be read from, instead use seqan3::aminoacid_alphabet.
+ * However, user-defined alphabets that want to model seqan3::aminoacid_alphabet need to make sure that it evaluates
+ * to `true` for their type.
  *
- * ### Customisation point
+ * ### Specialisation
+ *
+ * Do not specialise this trait directly. It acts as a wrapper and looks for two possible implementations
+ * (in this order):
+ *
+ *   1. A `static` member variable `enable_aminoacid` of the class `seqan3::custom::alphabet<your_type>`.
+ *   2. A free function `enable_aminoacid(your_type)` in the namespace of your type (or as `friend`).
+ *
+ * If none of these is found, the default value is defined as:
+ *
+ *   * `true` if the type inherits from seqan3::aminoacid_empty_base (or seqan3::aminoacid_base),
+ *   * `false` otherwise.
+ *
+ * Implementations of 1. and 2. are required to be marked `constexpr` and their (return) value must be convertible to
+ * `bool`.
+ * Implementations of 2. are required to be marked `noexcept`. The value passed to functions implementing 2.
+ * shall be ignored, it is only used for finding the function via argument-dependent lookup.
+ * In case that your type is not seqan3::is_constexpr_default_constructible_v and you wish to provide an implementation
+ * for 2., instead overload for `std::type_identity<your_type>`.
  *
- * If you define your own alphabet and want it to be recognised as an amino acid
- * alphabet by SeqAn, you need to specialise this type trait for your type and
- * have it inherit std::true_type.
+ * To make a type model seqan3::aminoacid_alphabet, it is recommend you derive from seqan3::aminoacid_base.
```

Comment:
```suggestion
 * To make a type model seqan3::aminoacid_alphabet, it is recommended that you derive from seqan3::aminoacid_base.
```

Reply:
Have you forgotten about this? Afterwards I can set the green light for this PR.

Reply:
Ah, yes, sorry.